pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/ReadableSequentialData.java

```diff
@@ -484,81 +453,16 @@ && readByte() < 0) {
-        long pos = position();
-        long limit = limit();
-        if (limit == pos) {
-            return readVarIntLongSlow(zigZag);
-        }
-        else if (limit - pos < 10) {
-            return readVarIntLongSlow(zigZag);
-        }
-
-        long x;
-        int y;
-        if ((y = readByte()) >= 0) {
-            return zigZag ? (y >>> 1) ^ -(y & 1) : y;
-        } else if ((y ^= (readByte() << 7)) < 0) {
-            x = y ^ (~0 << 7);
-        } else if ((y ^= (readByte() << 14)) >= 0) {
-            x = y ^ ((~0 << 7) ^ (~0 << 14));
-        } else if ((y ^= (readByte() << 21)) < 0) {
-            x = y ^ ((~0 << 7) ^ (~0 << 14) ^ (~0 << 21));
-        } else if ((x = y ^ ((long) readByte() << 28)) >= 0L) {
-            x ^= (~0L << 7) ^ (~0L << 14) ^ (~0L << 21) ^ (~0L << 28);
-        } else if ((x ^= ((long) readByte() << 35)) < 0L) {
-            x ^= (~0L << 7) ^ (~0L << 14) ^ (~0L << 21) ^ (~0L << 28) ^ (~0L << 35);
-        } else if ((x ^= ((long) readByte() << 42)) >= 0L) {
-            x ^= (~0L << 7) ^ (~0L << 14) ^ (~0L << 21) ^ (~0L << 28) ^ (~0L << 35) ^ (~0L << 42);
-        } else if ((x ^= ((long) readByte() << 49)) < 0L) {
-            x ^=
-                    (~0L << 7)
-                            ^ (~0L << 14)
-                            ^ (~0L << 21)
-                            ^ (~0L << 28)
-                            ^ (~0L << 35)
-                            ^ (~0L << 42)
-                            ^ (~0L << 49);
-        } else {
-            x ^= ((long) readByte() << 56);
-            x ^=
-                    (~0L << 7)
-                            ^ (~0L << 14)
-                            ^ (~0L << 21)
-                            ^ (~0L << 28)
-                            ^ (~0L << 35)
-                            ^ (~0L << 42)
-                            ^ (~0L << 49)
-                            ^ (~0L << 56);
-            if (x < 0L) {
-                if (readByte() < 0L) {
-                    throw new DataEncodingException("Malformed VarLong");
-                }
-            }
-        }
-        return zigZag ? (x >>> 1) ^ -(x & 1) : x;
-    }
+        long value = 0;
 
-    /**
-     * Read a 64bit protobuf varint at current {@link #position()}. A long var int can be 1 to 10 bytes.
-     * This is the slow path function.
-     *
-     * @return long read in var int format
-     * @param zigZag use protobuf zigZag varint encoding, optimized for negative numbers
-     * @throws BufferUnderflowException If the end of the sequence is reached before the final variable byte fragment
-     *                                  is read
-     * @throws DataAccessException if an I/O error occurs
-     * @throws DataEncodingException if the variable long cannot be decoded
-     */
-    default long readVarIntLongSlow(final boolean zigZag) {
-        long result = 0;
-        for (int shift = 0; shift < 64; shift += 7) {
+        for (int i = 0; i < 10; i++) {
```

Comment:
New implementation is faster than the old one according to benchmarks. Besides that, the old implementation was simply wrong (as you found yesterday and filed #143 about it).

The reason why there was code with those unwided if's is it was copied from google's implementation. However, google API is slightly different from ours, e.g. `CodedInputStream` doesn't update underlying buffer's position on reads, while our classes do. That's why code can't be copied as is, it needs to be adjusted anyway.

Reply:
Got it. Thanks for confirming that that this change improves the performance. It would be helpful to include this information in the description of the PR.

Reply:
Correct. Please, check benchmark results before and after this PR in the ticket comments, #67.

Reply:
Thanks for the pointer. The numbers look good to me.